test/gtest/ucm/cuda_hooks.cc

```diff
@@ -63,66 +25,145 @@ class cuda_hooks : public ucs::test {
             UCS_TEST_SKIP_R("can't init cuda device");
         }
 
-        ret = cuDeviceGet(&device, 0);
+        ret = cuDeviceGet(&m_device, 0);
         if (ret != CUDA_SUCCESS) {
             UCS_TEST_SKIP_R("can't get cuda device");
         }
 
-        ret = cuCtxCreate(&context, 0, device);
+        ret = cuCtxCreate(&m_context, 0, m_device);
         if (ret != CUDA_SUCCESS) {
             UCS_TEST_SKIP_R("can't create cuda context");
         }
 
-        memset(&alloc_event, 0, sizeof(alloc_event));
-        memset(&free_event, 0, sizeof(free_event));
+        /* Avoid memory allocation in event callbacks */
+        m_alloc_events.reserve(100);
```

Comment:
Testing PR on `ice02`, test `test_cuMemAllocAsync` seems stuck, backtrace shows allocation while `.push_back()`. Replacing `100` by `1000` fixes the issue.